packages/flutter_tools/lib/src/asset.dart

```diff
@@ -133,6 +136,9 @@ class ManifestAssetBundle implements AssetBundle {
   @override
   final Map<String, Map<String, DevFSContent>> deferredComponentsEntries = <String, Map<String, DevFSContent>>{};
 
+  @override
+  final List<File> inputFiles = <File>[];
```

Comment:
We can already create depfiles from an asset bundle, though it might not be exactly what you need: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/build_system/targets/assets.dart#L30

Did you look at using this method instead?

Reply:
Thanks for the pointer.

`copyAssets` looks sufficiently different from `buildAssets` [0] we use in `fuchsia_asset_builder`. Even if we can replace `buildAssets` with `copyAssets`, `fuchsia_asset_builder` writes a manifest file and considers that as the output in the depfile, which is also slightly different from the depfile from `copyAssets`.

The `Depfile` class looks pretty nice so I can switch to use that. Although not here since `AssetBundle` doesn't know about the manifest output.

WDYT?

[0] https://github.com/flutter/flutter/blob/a1fa62ad6f207281a11772f9155cbca4aaf3da8d/packages/flutter_tools/lib/src/bundle.dart#L164